src/xunit.analyzers/AssertThrowsShouldNotBeUsedForAsyncThrowsCheckFixer.cs

```diff
@@ -0,0 +1,156 @@
+            var invocation = root.FindNode(context.Span).FirstAncestorOrSelf<InvocationExpressionSyntax>();
+            var method = invocation.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+            var methodName = context.Diagnostics.First().Properties[AssertEqualShouldNotBeUsedForNullCheck.MethodName];
+            
+            var title = String.Format(TitleTemplate, methodName + "Async");
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title,
+                    createChangedDocument: ct => UseAsyncThrowsCheckAsync(context.Document, invocation, method, methodName + "Async", ct),
+                    equivalenceKey: title),
+                context.Diagnostics);
+        }
+
+        static async Task<Document> UseAsyncThrowsCheckAsync(Document document, InvocationExpressionSyntax invocation, MethodDeclarationSyntax method, string replacementMethod, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var memberAccess = (MemberAccessExpressionSyntax)invocation.Expression;
+            
+            var modifiers = GetModifiersWithAsyncKeywordAdded(method);
+            var returnType = await GetReturnType(method, invocation, document, cancellationToken);
+            var asyncThrowsInvocation = GetAsyncThrowsInvocation(invocation, replacementMethod, memberAccess);
+
+            editor.ReplaceNode(method,
+                method
+                    .ReplaceNode(invocation, asyncThrowsInvocation)
+                    .WithModifiers(modifiers)
+                    .WithReturnType(returnType));
+
+            return editor.GetChangedDocument();
+        }
+
+        private static SyntaxTokenList GetModifiersWithAsyncKeywordAdded(MethodDeclarationSyntax method)
+        {
+            return method.Modifiers.Any(SyntaxKind.AsyncKeyword)
+                ? method.Modifiers
+                : method.Modifiers.Add(SyntaxFactory.Token(SyntaxKind.AsyncKeyword));
+        }
+
+        private static async Task<TypeSyntax> GetReturnType(MethodDeclarationSyntax method, InvocationExpressionSyntax invocation, Document document, CancellationToken cancellationToken)
+        {
+            // Consider the case where a custom awaiter type is awaited
+            if (invocation.Parent.IsKind(SyntaxKind.AwaitExpression))
+                return method.ReturnType;
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodSymbol = semanticModel.GetSymbolInfo(method.ReturnType, cancellationToken).Symbol as ITypeSymbol;
+            var taskType = semanticModel.Compilation.GetTypeByMetadataName(typeof(Task).FullName);
+
+            if (taskType.IsAssignableFrom(methodSymbol))
+                return method.ReturnType;
+
+            return
```

Comment:
Thanks for the tip! I didn't know how to do that, so that's why I used the fully qualified namespace.